### PR TITLE
rpc: speedup get_outs.bin

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/cryptonote_core/blockchain.cpp [hunk 1]
```diff
@@ -1766,16 +1766,34 @@ bool Blockchain::get_outs(const COMMAND_RPC_GET_OUTPUTS_BIN::request& req, COMMA
...
     {
-      // get tx_hash, tx_out_index from DB
-      const output_data_t od = m_db->get_output_key(i.amount, i.index);
-      tx_out_index toi = m_db->get_output_tx_and_index(i.amount, i.index);
-      bool unlocked = is_tx_spendtime_unlocked(m_db->get_tx_unlock_time(toi.first));
+      amounts.push_back(i.amount);
+      offsets.push_back(i.index);
+    }
+    m_db->get_output_key(amounts, offsets, data);
+    if (data.size() != req.outputs.size())
```
Comment (xiphon) [inline src/cryptonote_core/blockchain.cpp]: `allow_partial` is false by default, will never satisfy the condition then.

File: src/blockchain_db/lmdb/db_lmdb.h [hunk 2]
```diff
@@ -243,7 +243,7 @@ class BlockchainLMDB : public BlockchainDB
   virtual uint64_t get_num_outputs(const uint64_t& amount) const;
 
   virtual output_data_t get_output_key(const uint64_t& amount, const uint64_t& index);
-  virtual void get_output_key(const uint64_t &amount, const std::vector<uint64_t> &offsets, std::vector<output_data_t> &outputs, bool allow_partial = false);
+  virtual void get_output_key(const std::vector<uint64_t> &amounts, const std::vector<uint64_t> &offsets, std::vector<output_data_t> &outputs, bool allow_partial = false);
```
Comment (vtnerd) [inline src/blockchain_db/lmdb/db_lmdb.h]: If `amounts` were a `epee::span<const uint64_t>` then the caller wouldn't have to allocate a vector for the one element case - a one element array would suffice.

Should the one element behavior of `amounts` be documented in the header now?

File: src/rpc/core_rpc_server_commands_defs.h [hunk 3]
```diff
@@ -50,7 +50,7 @@ namespace cryptonote
 // advance which version they will stop working with
 // Don't go over 32767 for any of these
 #define CORE_RPC_VERSION_MAJOR 2
-#define CORE_RPC_VERSION_MINOR 1
+#define CORE_RPC_VERSION_MINOR 2
```
Comment (vtnerd) [inline src/rpc/core_rpc_server_commands_defs.h]: What is the minor version used for? I think older clients are still compatible, as the receiver would just assume that `get_txid == true` (existing behavior).

**Comment by moneromooo-monero** [comment]
IIRC, when all was cached, it was about two to three times as fast.

File: src/rpc/core_rpc_server_commands_defs.h (hunk 3 above)
Comment (moneromooo-monero) [reply src/rpc/core_rpc_server_commands_defs.h]: It gets bumped if the changes are backward compatible. They are backward compatible here as you say. A client can now tell whether some particular thing (here, get_txid) is in the daemon it's talking to.

File: src/blockchain_db/lmdb/db_lmdb.h (hunk 2 above)
Comment (moneromooo-monero) [reply src/blockchain_db/lmdb/db_lmdb.h]: Yes. It is.

**Pull request merged**